### PR TITLE
Load genes static files

### PR DESCRIPTION
**Pull request by moonso**
This PR adds a functionality or fixes a bug.

**How to test**:
1. Install this on your local machine
1. If you like you can test the new `scout download` command
1. Run `scout setup database` and point to either the files downloaded in the step above or point to the ones in `scout/demo/resources`

**Expected outcome**:
The database should be able to setup by using static files

**Review:**
- [x] code approved by CR
- [x] tests executed by @parlar 🥇

### CODE REVIEW

**Comment by moonso** [comment]
@parlar please try this out

File: scout/demo/resources/__init__.py [hunk 1]
```diff
@@ -3,54 +3,81 @@
...
 # Gene sources:
-hgnc_file = 'demo/resources/hgnc_reduced_set.txt'
-exac_file = 'demo/resources/forweb_cleaned_exac_r03_march16_z_data_pLI_reduced.txt'
-transcripts37_file = 'demo/resources/ensembl_transcripts_37_reduced.txt'
-genes37_file = 'demo/resources/ensembl_genes_37_reduced.txt'
-exons37_file = 'demo/resources/ensembl_exons_37_reduced.txt'
-transcripts38_file = 'demo/resources/ensembl_transcripts_38_reduced.txt'
-genes38_file = 'demo/resources/ensembl_genes_38_reduced.txt'
-exons38_file = 'demo/resources/ensembl_exons_38_reduced.txt'
+hgnc_file = "demo/resources/hgnc_reduced_set.txt"
```
Comment (northwestwitch) [inline scout/demo/resources/__init__.py]: Really??? :D

Comment (moonso) [reply scout/demo/resources/__init__.py]: What?

Comment (northwestwitch) [reply scout/demo/resources/__init__.py]: That it's blacked!

Comment (moonso) [reply scout/demo/resources/__init__.py]: Haha yes

**Comment by moonso** [comment]
@northwestwitch and @dnil can we try to merge this?

**Comment by dnil** [comment]
Resolve conflicts? Or wait for #1668?

**Comment by moonso** [comment]
Conflicts solved!

**Comment by northwestwitch** [comment]
Reviewing this now!

**Comment by parlar** [comment]
Sorry, missed this. Thanks! WIll try immediately!

**Comment by parlar** [comment]
Download works fine. Adding --api-key works for dl of omim.  When it comes to setting up the db I feel stupid, however. 

How do I actually tell scout setup to use the downloaded resources for setting up?

* I cannot find any option to tell scout to use local resources
* ```scout setup database <path/to/resource/folder>``` didn't work
* executing ```scout setup database``` in the resource folder did not work either
* neither did moving the downloaded files to  ```scout/resources```

**Comment by parlar** [comment]
Figured it out by inspecting ```scout/commands/setup/setup_scout.py``` . The option ```--files <path>``` must be added. 

Now it works!

May I suggest adding this to the CLI help?

**Comment by moonso** [comment]
Hi @parlar , thanks for testing and suggestions. I've added a small help text to the `--files` option, however the `--api-key` that you refer to have help texts everywhere, at least where I looked. Please point to where it is missing. 

So @dnil , @northwestwitch this is ready for review.

**Comment by northwestwitch** [comment]
I can review!

File: scout/commands/download/hgnc.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+def print_hgnc(out_dir):
+    """Print HPO files to a directory
+
+    Args:
+        out_dir(Path)
+    """
+    file_name = "hgnc.txt"
+    file_path = out_dir / file_name
+    LOG.info("Print HGNC genes to %s", file_path)
```
Comment (northwestwitch) [inline scout/commands/download/hgnc.py]: Download genes to path

File: scout/commands/download/hpo.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+def print_hpo(out_dir):
+    """Print HPO files to a directory
+
+    Args:
+        out_dir(Path)
+    """
+    hpo_file_name = "hpo.obo"
+    hpo_file_path = out_dir / hpo_file_name
+    LOG.info("Print HPO terms to %s", hpo_file_path)
```
Comment (northwestwitch) [inline scout/commands/download/hpo.py]:
```suggestion
    LOG.info("Download HPO terms to %s", hpo_file_path)
```

File: scout/commands/download/hpo.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    with hpo_file_path.open("w", encoding="utf-8") as outfile:
+        for line in fetch_hpo_terms():
+            outfile.write(line + "\n")
+
+    hpo_file_name = "ALL_SOURCES_ALL_FREQUENCIES_genes_to_phenotype.txt"
+    hpo_file_path = out_dir / hpo_file_name
+    LOG.info("Print HPO genes to %s", hpo_file_path)
```
Comment (northwestwitch) [inline scout/commands/download/hpo.py]:
```suggestion
    LOG.info("Download HPO genes to %s", hpo_file_path)
```

File: scout/commands/download/hpo.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    with hpo_file_path.open("w", encoding="utf-8") as outfile:
+        for line in fetch_hpo_genes():
+            outfile.write(line + "\n")
+
+    hpo_file_name = "ALL_SOURCES_ALL_FREQUENCIES_phenotype_to_genes.txt"
+    hpo_file_path = out_dir / hpo_file_name
+    LOG.info("Print HPO TO genes to %s", hpo_file_path)
```
Comment (northwestwitch) [inline scout/commands/download/hpo.py]:
```suggestion
    LOG.info("Download HPO TO genes to %s", hpo_file_path)
```

File: scout/commands/download/hpo.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    with hpo_file_path.open("w", encoding="utf-8") as outfile:
+        for line in fetch_hpo_to_genes():
+            outfile.write(line + "\n")
+
+    hpo_file_name = "ALL_SOURCES_ALL_FREQUENCIES_diseases_to_genes_to_phenotypes.txt"
+    hpo_file_path = out_dir / hpo_file_name
+    LOG.info("Print HPO disease %s", hpo_file_path)
```
Comment (northwestwitch) [inline scout/commands/download/hpo.py]:
```suggestion
    LOG.info("Download HPO disease %s", hpo_file_path)
```

File: scout/commands/download/omim.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+def print_omim(out_dir, api_key):
+    """Print HPO files to a directory
+
+    Args:
+        out_dir(Path)
+    """
+    mim_files = fetch_mim_files(api_key, mim2genes=True, genemap2=True)
+    file_name = "genemap2.txt"
+    file_path = out_dir / file_name
+    LOG.info("Print genemap genes to %s", file_path)
```
Comment (northwestwitch) [inline scout/commands/download/omim.py]:
```suggestion
    LOG.info("Download genemap genes to %s", file_path)
```

File: scout/commands/setup/setup_scout.py [hunk 8]
```diff
@@ -137,25 +251,26 @@ def setup(context, institute, user_mail, user_name):
         "user_mail": user_mail,
     }
 
-    mongodb = current_app.config["MONGO_DBNAME"]
+    mongodb_name = current_app.config["MONGO_DBNAME"]
```
Comment (northwestwitch) [inline scout/commands/setup/setup_scout.py]: 👍

File: scout/parse/omim.py [hunk 9]
```diff
@@ -38,6 +38,67 @@ def parse_omim_line(line, header):
...
+
+    If a special symbol is used in the description it indicates the disease status.
+
+    """
+    parsed_phenotypes = []
+
+    for phenotype_info in phenotype_entry.split(";"):
+        if not phenotype_info:
+            continue
+        phenotype_info = phenotype_info.lstrip()
```
Comment (northwestwitch) [inline scout/parse/omim.py]: Why not a simple strip?

File: scout/commands/download/ensembl.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    for build in builds:
+        file_path = out_dir / file_name.format(build)
+        LOG.info("Print ensembl info %s to %s", build, file_path)
+        with file_path.open("w", encoding="utf-8") as outfile:
+            for line in fetch_function(build=build):
+                outfile.write(line + "\n")
+        LOG.info("Ensembl info printed")
+
+
+def print_ensembl_genes(out_dir, build=None):
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/commands/download/ensembl.py]: Good suggestion, I fixed that

File: scout/parse/omim.py (hunk 9 above)
Comment (moonso) [reply scout/parse/omim.py]: Guess I only want to strip on the left side but it might not matter

**Comment by moonso** [comment]
Fixed according to your suggestions @northwestwitch and added tests

**Comment by northwestwitch** [comment]
> Fixed according to your suggestions @northwestwitch and added tests

I'll check after lunch!

**Comment by northwestwitch** [comment]
Good job, also with the tests!! 👍 I think it's fine to merge, whenever you are ready!

**Pull request merged**